scripts/test/Muon/loadfile_presenter_test.py

```diff
@@ -0,0 +1,510 @@
+        self.mock_user_input_text("C:/dir1/file1.nxs;C:/dir2/file2.nxs")
+        self.mock_model_to_load_workspaces([[1], [2]], [1234, 1235])
+
+        self.presenter.handle_file_changed_by_user()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.view.warning_popup.call_count, 1)
+
+    @run_test_with_and_without_threading
+    def test_single_file_from_browse_loaded_into_model_and_view_in_single_file_mode(self):
+        self.mock_browse_button_to_return_files(["C:/dir1/file1.nxs"])
+        self.mock_model_to_load_workspaces([[1]], [1234])
+        self.view.set_file_edit = mock.Mock()
+
+        self.presenter.on_browse_button_clicked()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.model.loaded_filenames, ["C:/dir1/file1.nxs"])
+        self.assertEqual(self.model.loaded_workspaces, [[1]])
+        self.assertEqual(self.model.loaded_runs, [1234])
+
+        self.view.set_file_edit.assert_called_once_with("C:/dir1/file1.nxs", mock.ANY)
+
+    @run_test_with_and_without_threading
+    def test_single_file_from_user_input_loaded_into_model_and_view_in_single_file_mode(self):
+        self.view.set_file_edit = mock.Mock()
+        self.mock_model_to_load_workspaces([[1]], [1234])
+        self.mock_user_input_text("C:/dir1/file1.nxs")
+
+        self.presenter.handle_file_changed_by_user()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.model.loaded_filenames, ["C:/dir1/file1.nxs"])
+        self.assertEqual(self.model.loaded_workspaces, [[1]])
+        self.assertEqual(self.model.loaded_runs, [1234])
+
+        self.view.set_file_edit.assert_called_once_with("C:/dir1/file1.nxs", mock.ANY)
+
+    @run_test_with_and_without_threading
+    def test_that_if_invalid_file_selected_in_browser_view_does_not_change(self):
+        self.mock_browse_button_to_return_files(["not_a_file"])
+        self.mock_model_to_load_workspaces([[1]], [1234])
+
+        self.view.set_file_edit = mock.Mock()
+        self.view.reset_edit_to_cached_value = mock.Mock()
+
+        self.presenter.on_browse_button_clicked()
+        self.Runner(self.presenter._load_thread)
+
+        self.model.load_workspace_from_filename = mock.Mock(side_effect=self.load_failure)
+
+        cc = self.view.set_file_edit.call_count
+        self.presenter.on_browse_button_clicked()
+        self.Runner(self.presenter._load_thread)
+
+        self.assertEqual(self.view.set_file_edit.call_count, cc)
+        self.assertEqual(self.view.reset_edit_to_cached_value.call_count, 0)
+
+    @run_test_with_and_without_threading
+    def test_that_view_reverts_to_previous_text_if_users_supplies_invalid_text(self):
+        self.load_workspaces_into_model_and_view_from_browse([[1]], [1234], ["C:/dir1/EMU0001234.nxs"])
+
+        invalid_user_input = ["some random text", "1+1=2", "..."]
+
+        call_count = self.view.reset_edit_to_cached_value.call_count
```

Comment:
I think this should be zero.

Reply:
The value of `call_count` here is not relevant; it is being used as a counter for the for loop on the next line; all we test is that `call_count` increments by `1` on each iteration.